### DISCUSSION TITLE
IP based DNS

### DISCUSSION THREAD

**Discussion by TimVeen** [qdm12/gluetun#1077] [Q&A] [answered]
So currently it is possible to preset a DNS, however surfshark, for example, has its own DNS server. Is it possible to automatically change the DNS address to the given IP address? Because this is what surfshark can do

**Comment by frepke** [comment, marked as answer]
Hello TimVeen,

When you set `- DOT=off` and `- DNS_PLAINTEXT_ADDRESS=162.252.172.57` you can check the result on [https://ipleak.net/](https://ipleak.net/). Then your ip and dns are the same address.

ps
I've setup a firefox container in the Gluetun stack.

**Comment by jcheroske** [comment]
Is it possible to set a specific DoT address somehow, and not specify a provider?

**Reply by qdm12** [reply]
Subscribe to #137 when this is closed this will be possible.

I'm working on https://github.com/qdm12/dns/tree/v2.0.0-beta still, which will be used in Gluetun soon.
